File: junkz/playstruct.c
// playstruct.c

// A complete C Program
#include <stdio.h>
#include <stdlib.h>
#include "playstruct.h"

typedef struct complex {
    double real;
    double imag;
} *Complex;

Complex new_cmplx(double real, double imag) {
    Complex cmplx = malloc(sizeof(struct complex));
    if (!cmplx) {
        perror("Insufficient Memory Allocation.");
        exit(EXIT_FAILURE);
    }
    cmplx->real = real;
    cmplx->imag = imag;
    return cmplx;
}

Complex new_cmplx_default() {
    return new_cmplx(0.0, 0.0);
}

void print_cmplx(Complex cmplx) {
    printf("Complex(real: %.2f, imag: %.2fi)\n", cmplx->real, cmplx->imag);
}

void free_cmplx(Complex cmplx) {
    free(cmplx);
}

File: junkz/playstruct.rs
#![allow(dead_code, unused)]

#[derive(Default, Copy, Clone)]
struct Complex {
    real: f64,
    imag: f64,
}

impl Complex {
    fn new() -> Self {
        Self::default()
    }

    fn new_with_values(real: f64, imag: f64) -> Self {
        Self { real, imag }
    }

    fn print(&self) {
        println!("Complex(real: {:.2}, imag: {:.2}i)", self.real, self.imag);
    }
}

fn main() {
    let cmplx = Complex::new();
    cmplx.print();

    let cmplx2 = Complex::new_with_values(3.25, 7.1);
    cmplx2.print();
}
